great_expectations/datasource/fluent/data_asset/path/path_data_asset.py

```diff
@@ -239,3 +222,9 @@ def _get_reader_method(self) -> str:
     def _get_reader_options_include(self) -> set[str]:
         # subtypes control how reader options get serialized
         raise NotImplementedError
+
+    def _get_sortable_partitioner(
+        self, partitioner: Optional[PartitionerT]
+    ) -> Optional[PartitionerSortingProtocol]:
+        # allow subclasses to determine sorting configuration.
```

Comment:
Can this be an `abstractmethod`?

Reply:
In this case, yes ([e89cf57](https://github.com/great-expectations/great_expectations/pull/9888/commits/e89cf570efcd5e31783fa106f40a2bd71ea09345)) - but that makes me wonder why do we have a handful of other `NotImplemented` methods on this class, can we make them `abstractmethod`s as well? And the answer is no, we can't, because [Pandas dynamic assets](https://github.com/great-expectations/great_expectations/blob/86a58627cc09ec8ddde4028dd4d83024d82a8e25/great_expectations/datasource/fluent/data_asset/path/pandas/generated_assets.py#L35) are given the type `FileAsset`, but their implementations of these methods are dynamically generated, so mypy fails.

Reply:
oh wow good to know, thanks for digging in!